Tools/LyTestTools/ly_test_tools/_internal/managers/abstract_resource_locator.py

```diff
@@ -60,7 +60,15 @@ def _find_project_json(engine_root, project):
     if os.path.isfile(manifest_json):
         # Read the o3de_manifest.json 
         with open(manifest_json, "r") as manifest_file:
-            json_data = json.load(manifest_file)
+
+            # Try to load the contents of the o3de_manifest.json file
+            try:
+                manifest_file_contents = manifest_file.read()
+                json_data = json.loads(manifest_file_contents)
+            except json.JSONDecodeError as jderror:
+                logger.error(f"Attempted to open the contents of {manifest_json}, found invalid JSON: {manifest_file_contents}")
+                raise jderror
```

Comment:
There are two other ways of finding the project json after attempting to read the manifest so I wouldn't mind not raising here and allowing the other attempts to try

Reply:
That's an interesting point, I hadn't looked further down in this file.

Invalid contents is technically different from not finding the intended file, so there's an argument about not immediately raising and instead logging a warning. I think this JSONDecodeError can be handled in the same try-block that is handling KeyError below, and similarly log the error. The function can then be left to raise that it could not find the manifest as it already does, with the warning as a breadcrumb.

On the downside, this makes it less immediately obvious to see that an error was encountered in the primary place that O3DE project information is stored.

I would slightly prefer a warning-then-raise, but not strongly enough to revoke my approval.